zerver/views/webhooks/trello/card_actions.py

```
@@ -0,0 +1,165 @@
+REOPEN = 'reopenCard'
+SET_DUE_DATE = 'setDueDate'
+CHANGE_DUE_DATE = 'changeDueDate'
+REMOVE_DUE_DATE = 'removeDueDate'
+ADD_LABEL = 'addLabelToCard'
+REMOVE_LABEL = 'removeLabelFromCard'
+ADD_MEMBER = 'addMemberToCard'
+REMOVE_MEMBER = 'removeMemberFromCard'
+
+
+TRELLO_SUBJECT_TEMPLATE = '[{board_name}]"{card_name}".'
+TRELLO_MESSAGE_TEMPLATE = '{full_name} {rest}.'
+
+
+ACTIONS_TO_MESSAGE_MAPPER = {
+    CREATE: 'created the card',
+    CHANGE_LIST: 'moved "{card_name}" from {old_list} to {new_list}',
+    CHANGE_NAME: 'renamed the card from "{old_name}" to "{new_name}"',
+    ARCHIVE: 'archived the card',
+    REOPEN: 'reopened the card',
+    SET_DUE_DATE: 'set due date to {due_date}',
+    CHANGE_DUE_DATE: 'changed due date to {due_date}',
+    REMOVE_DUE_DATE: 'removed due date',
+    ADD_LABEL: 'added a {color} label with \"{text}\"',
+    REMOVE_LABEL: 'removed a {color} label with \"{text}\"',
+    ADD_MEMBER: 'added {member_name} to the card',
+    REMOVE_MEMBER: 'removed {member_name} from the card',
+}
+
+
+def process_card_action(payload, action_type):
+    action_type = get_proper_action(payload, action_type)
+    return get_subject(payload), get_body(payload, action_type)
+
+
+def get_proper_action(payload, action_type):
+    if action_type == 'updateCard':
+        data = get_action_data(payload)
+        if data.get('listBefore'):
+            return CHANGE_LIST
+        if data.get('old').get('name'):
+            return CHANGE_NAME
+        if data.get('old').get('due', False) is None:
+            return SET_DUE_DATE
+        if data.get('old').get('due'):
+            if data.get('card').get('due', False) is None:
+                return REMOVE_DUE_DATE
+            else:
+                return CHANGE_DUE_DATE
+        if data.get('old').get('closed') is False and data.get('card').get('closed'):
+            return ARCHIVE
+        if data.get('old').get('closed') and data.get('card').get('closed') is False:
+            return REOPEN
+
```

Probably should raise an exception in the else case here.